react/Carousel.tsx

```diff
@@ -290,71 +90,255 @@ export default class Carousel extends Component<Props, State> {
+              default: false,
+              isLayout: false,
+              title: 'admin/editor.carousel.banner.externalRoute.title',
+              type: 'boolean',
+            },
+          },
+          dependencies: {
+            externalRoute: {
+              oneOf: [
+                {
+                  properties: {
+                    externalRoute: {
+                      enum: [false],
+                    },
+                    ...internalRouteSchema,
+                  },
+                },
+                {
+                  properties: {
+                    externalRoute: {
+                      enum: [true],
+                    },
+                    ...externalRouteSchema,
+                  },
+                },
+              ],
+            },
+          },
+          title: 'admin/editor.carousel.banner.title',
+          type: 'object',
+        },
+        minItems: 1,
+        title: 'admin/editor.carousel.banners.title',
+        type: 'array',
+      },
+      height: {
+        default: 420,
+        enum: [420, 440],
+        isLayout: true,
+        title: 'admin/editor.carousel.height.title',
+        type: 'number',
+      },
+      showArrows: {
+        default: true,
+        isLayout: true,
+        title: 'admin/editor.carousel.showArrows.title',
+        type: 'boolean',
+      },
+      showDots: {
+        default: true,
+        isLayout: true,
+        title: 'admin/editor.carousel.showDots.title',
+        type: 'boolean',
+      },
+      autoplay: {
+        default: true,
+        isLayout: false,
+        title: 'admin/editor.carousel.autoplay.title',
+        type: 'boolean',
+      },
+    },
+    title: 'admin/editor.carousel.title',
+    type: 'object',
   }
 }
+
+Carousel.propTypes = {
+  /** Should change images automatically */
+  autoplay: PropTypes.bool.isRequired,
+  /** How long it should wait to change the banner in secs */
+  autoplaySpeed: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  /** Banners that will be displayed by the Carousel */
+  banners: PropTypes.arrayOf(
+    PropTypes.shape({
+      /** The url where the image is pointing to, in case of internal route */
+      customInternalURL: PropTypes.string,
+      /** The description of the image */
+      description: PropTypes.string,
+      /** The image url of the banner */
+      image: PropTypes.string,
+      /** The page where the image is pointing to */
+      page: PropTypes.string,
+      /** Params of the url */
+      params: PropTypes.string,
+      /** Indicates if the route is external or internal */
+      typeOfRoute: PropTypes.string,
+      /** The url where the image is pointing to, in case of external route */
+      url: PropTypes.string,
+    })
+  ),
+  /** Max height size of the banners */
+  height: PropTypes.number.isRequired,
+  /** Set visibility of arrows */
+  showArrows: PropTypes.bool,
+  /** Set visibility of dots */
+  showDots: PropTypes.bool,
+}
```

Comment:
You could remove this. Since this is already TS and these `propTypes` aaaalways get outdated and just generate errors/warnings in the console.
I know that `propTypes` are more of a `runtime` verification, but I think they are more inconvenient than useful to us when using TypeScript